nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/test/java/org/apache/nifi/elasticsearch/integration/ElasticSearchClientService_IT.java

```diff
@@ -0,0 +1,745 @@
+    private ElasticSearchClientServiceImpl service;
+
+    static final String INDEX = "messages";
+    public static String TYPE;
+
+    static final ComparableVersion VERSION = new ComparableVersion(System.getProperty("es_version", "0.0.0"));
+    static final ComparableVersion ES_7_10 = new ComparableVersion("7.10");
+    static final ComparableVersion ES_8_0 = new ComparableVersion("8.0");
+
+    static final String FLAVOUR = System.getProperty("es_flavour");
+    static final String DEFAULT = "default";
+
+    private static TlsConfiguration generatedTlsConfiguration;
+    private static TlsConfiguration truststoreTlsConfiguration;
+
+    @BeforeAll
+    static void beforeAll() throws IOException {
+        TYPE = getElasticMajorVersion() == 6 ? "_doc" : "";
+        System.out.println(
+                String.format("%n%n%n%n%n%n%n%n%n%n%n%n%n%n%nTYPE: %s%n%s%s%n%n%n%n%n%n%n%n%n%n%n%n%n%n%n",
+                        TYPE, IMAGE.getRepository(), IMAGE.getVersionPart())
+        );
+
+        generatedTlsConfiguration = new TemporaryKeyStoreBuilder().build();
+        truststoreTlsConfiguration = new StandardTlsConfiguration(
+                null,
+                null,
+                null,
+                generatedTlsConfiguration.getTruststorePath(),
+                generatedTlsConfiguration.getTruststorePassword(),
+                generatedTlsConfiguration.getTruststoreType()
+        );
+
+        setupTestData();
+    }
+
+    @BeforeEach
+    void before() throws Exception {
+        runner = TestRunners.newTestRunner(TestControllerServiceProcessor.class);
+        service = new ElasticSearchClientServiceImpl();
+        runner.addControllerService("Client Service", service);
+        runner.setProperty(service, ElasticSearchClientService.HTTP_HOSTS, String.format("http://%s", ELASTICSEARCH_CONTAINER.getHttpHostAddress()));
+        runner.setProperty(service, ElasticSearchClientService.CONNECT_TIMEOUT, "10000");
+        runner.setProperty(service, ElasticSearchClientService.SOCKET_TIMEOUT, "60000");
+        runner.setProperty(service, ElasticSearchClientService.RETRY_TIMEOUT, "60000");
+        runner.setProperty(service, ElasticSearchClientService.SUPPRESS_NULLS, ElasticSearchClientService.ALWAYS_SUPPRESS.getValue());
+        runner.setProperty(service, ElasticSearchClientService.USERNAME, "elastic");
+        runner.setProperty(service, ElasticSearchClientService.PASSWORD, "s3cret");
+
+        try {
+            runner.enableControllerService(service);
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            throw ex;
+        }
+
+        service.refresh(null, null);
+    }
+
+    @AfterEach
+    void after() throws Exception {
+        service.onDisabled();
+    }
+
+    @Test
+    public void bootstrapTest() {
+
+    }
+
+    private String prettyJson(Object o) throws JsonProcessingException {
+        return new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(o);
```

Comment:
Suggest defining a class-level ObjectMapper and set it up before all tests are run - creating the ObjectMapper is a relatively expensive operation

Accept this a test class and not production code, but still good to take simple steps for performance

Reply:
Done.